pkg/storage/sqlstorage/balances_test.go

```diff
@@ -0,0 +1,86 @@
+package sqlstorage
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/numary/ledger/pkg/storage"
+	"github.com/pborman/uuid"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestBalances(t *testing.T) {
```

Comment:
- It would be better to use a new test case in the file `store_test.go` where all the other Store methods are tested. Also, you are here only testing with SQLite which is not enough.
- Pagination is not tested.
- Actual results with data is not tested. I think that the unit test below is not enough.
- The tests for `GetBalancesAggregated` are missing.

Reply:
all those tests are done in the controller part, do you really want to do them again here ?

I've added this specifically for SQLite because i'm doing a complex request to mimic the postgres result (only in the getBalances route, this is why you don't see GetBalancesAggregated).

If you think we also need to do the unit tests in sqlstorage, i think we should start using mocks in the controller, and do real unit tests on each package ? But i'd rather we take this decision outside of this PR

@gfyrag do you have any inputs ?

Reply:
If we follow the same rules as other tests, so yes, the comments of @antoinegelloz make sense, we should have all tests inside store_test.go. The setup of the test will tests sqlite as well as postgres so it will be a super set of your actual tests. 
We could mock storage on controllers, but i think the benefits are weak. @jdupas22 What benefits you see?
I'm agree, using concrete implementation of the storage inside controllers duplicates tests from storage, but force us to use a mock just to avoid the duplication seems counter productive for me.
Actually, sqlite storage can be considered as a mock as it is not an external service. So it is easily configurable inside tests.
It gives more surfaces without big cost.

Reply:
the only benefit i see is less tests by packages, and easier scopes for each packages to test. I'm not saying it's better than what we have now, it's just not the same kind of tests, hence the weak benefits.

I can add the postgres tests here alongside what i've already done for sqlite, because it tests a specific function (even though already tested in controllers) that is very critical to the endpoint, but I think adding other tests here would be a waste of time